plugins/catalog-backend/src/service/NextEntitiesCatalog.ts

```diff
@@ -131,29 +131,25 @@ function parseFilter(
   db: Knex,
 ): Knex.QueryBuilder {
   if (isEntitiesSearchFilter(filter)) {
-    return query.where(function filterFunction() {
+    return query.andWhere(function filterFunction() {
       addCondition(this, db, filter);
     });
   }
 
   if (isOrEntityFilter(filter)) {
-    let cumulativeQuery = query;
-    for (const subFilter of filter.anyOf ?? []) {
-      cumulativeQuery = cumulativeQuery.orWhere(subQuery =>
```

Comment:
I believe this is the root of the bug - previously, we were chaining `orWhere`s onto the top-level query when the filter was an "OR" filter. This means that if `parseFilter` were called multiple times as in the example below, we'd end up joining the clauses from the filters incorrectly - in the example below, ending up with `(A && B && C) || D || E` instead of `(A && B && C) && (D || E)`.

```typescript
filter = parseFilter({ allOf: [ A, B, C ] }, filter, db);
filter = parseFilter({ anyOf: [ D, E ] }, filter, db);
```